File: backend/springboot/src/main/java/tn/maiko26/springboot/models/TeamMember.java
package tn.maiko26.springboot.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Data
@Table(name = "team_members")
public class TeamMember {

    @Id
    @Column(name = "team_member_id")
    private String id;

    @ManyToOne // Many team members belong to one team
    @JoinColumn(name = "team_id", nullable = false) // Defines the foreign key column
    private Team team;

    @ManyToOne // Many team members belong to one user (email is unique in your system)
    @JoinColumn(name = "email", nullable = false) // Defines the foreign key column
    private User user;

    @Column(name = "role", columnDefinition = "varchar(255) default 'user'")
    private String role;
}

File: backend/dotnet/Models/relations/TeamMember.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace dotnet.Models
{
    // TeamMember.cs
    [Table("team_members")]
    public class TeamMember
    {
        [Key, Column("team_id")]
        public int TeamId { get; set; }

        [Key, Column("email")]
        public string Email { get; set; } = string.Empty;

        // Navigation properties
        [JsonIgnore]
        public Team Team { get; set; } = null!;
        [JsonIgnore]
        public User User { get; set; } = null!;
    }
}